### ISSUE TITLE
Encoding Error with ¿ character becoming Â¿ on Spoke Backend

### ISSUE THREAD

**Issue by bengilvar** [politics-rewired/Spoke#1595]
**Your issue may already be reported!**
Please search on the [issue tracker](../) before creating one.

- [YES] I have searched through existing issues and did not find an existing report

**Describe the bug**
A client wrote the following message, which from what I can tell is in GSM encoding: Hola {firstName}, soy Felipe y vivo en Redlands. Quieren prohibir vendedores ambulantes en Redlands. ¿Apoya protecciones para los vendedores y sus familias?

This is GSM according to both the [twillio message segment ](https://twiliodeved.github.io/message-segment-calculator/)estimator and spoke (see picture). However, the client was charged for 3 segments instead of one and we saw 3 segments on the backend (see attached csv) However, although we see "¿" in the spoke display, on the backend we see "Â¿" The Â seems to be making this not GSM as ¿ is a GSM character. See GSM characters: https://en.wikipedia.org/wiki/GSM_03.38#GSM_7-bit_default_alphabet_and_extension_table_of_3GPP_TS_23.038_.2F_GSM_03.38 

**To Reproduce**

1.  Go to a Spoke campaign
2.  Copy the above text into the interactions section for an initial message
3. Send the message
4.  See # of segments on the backend

**Expected behavior**
This message sends with 1 segment as expected, because all characters are GSM characters

**Screenshots**
![GSM](https://user-images.githubusercontent.com/102557290/236319529-e6c3ab36-283d-4e85-9d90-333721c7b981.png)

**Desktop (please complete the following information):**

- Browser [chrome, i]

[iedo query.csv](https://github.com/politics-rewired/Spoke/files/11401193/iedo.query.csv)

**Additional context**

**Comment by mkoontz-rewired** [comment]
@ajohn25 and I tested the message in the issue with both Telnyx and Bandwidth on staging.
We copied the same message: `Hola {firstName}, soy Felipe y vivo en Redlands. Quieren prohibir vendedores ambulantes en Redlands. ¿Apoya protecciones para los vendedores y sus familias?` from a notion page when creating campaigns to test.

**Bandwidth test:**

@ajohn25 was nice enough to contact support and confirm that the message we sent containing the inverted question mark was calculated as 1 segment by Bandwidth.

**Telnyx test:**

After misunderstanding the issue description (I thought testing with the Twilio message segment estimator meant this user was sending over Twilio), I tested the message with Telnyx, the  _actual_ telecom provider used by this user.
The delivery report confirmed that Telnyx calculated the message as 3 segments.
Telnyx's definition of valid GSM-7 characters is different from Twilio's segment calculator:

```
{
    "0123456789"
    "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    "abcdefghijklmnopqrstuvwxyz"
    "\n\r !\"#\$%&'()*+,-./:;<=>?@[\\]^_{}|~"
}
``` 

from https://developers.telnyx.com/docs/v2/messaging/messages/resources/configuration_and_limitations/character_and_rate_limits/

So they're accurately charging us for _their_ definition of valid GSM-7 characters. 

**Open Question:**

**Because we're moving most traffic to Bandwidth, the question is whether or not it's worth it at this time to update the client-side estimator to calculate estimated segments/encoding used based on what messaging service is being used.** 
 @ajohn25 thinks it's probably not worth it, but wondering if y'all (@hiemanshu, @bchrobot) agree.
 
 If it is worth it, I'll go ahead and make fixing that part of this ticket. If it's not, I can either create a ticket for it and close this one or just close as is.
